GraphEngine.Jit: incorporate asmjit

yatli opened the pull request.

Commit:
GraphEngine.Jit: incorporate asmjit

Commit:
wip

Commit:
work in progress @ Philippine

thautwarm committed:
codegen

thautwarm committed:
utilities and the spec of implementations

thautwarm reviewed (approved):
Is asmjit the way to dynamically generate our data access and manipulation methods?

Commit:
GraphEngine.Jit: asmjit binding builds successfully. build script not working, build manually with VS.

Commit:
GraphEngine.Jit: add JitRuntime binding

thautwarm committed:
implement-code-gen(unfinished)

Commit:
GraphEngine.Jit: fix build script; GraphEngine.Jit.Native now only include CppSharp as assembly, no build scripts

thautwarm committed:
implementations of ffi method code-gen(unfinished)

thautwarm committed:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit

Commit:
Verb: add input/output specs

Commit:
Merge branch 'jit' of github.com:Microsoft/GraphEngine into jit

Commit:
...

Commit:
fixing build

Commit:
GraphEngine.Jit: refactoring

thautwarm committed:
...

thautwarm committed:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit

Commit:
GraphEngine.Jit: allow composing FuncSignature manually

thautwarm committed:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit

thautwarm committed:
FFI.SwigGen: CodeGenerator implemented

thautwarm committed:
FFI.SwigGen: review Gen.fs

thautwarm committed:
FFI.SwigGen: Gen.fs. refactor, generateCodeGen function returns a sequence of codegenerator with verb

thautwarm committed:
FFI.SwigGen: Gen.fs, refactored. implement

thautwarm committed:
FFI.SwigGen: Gen.fs, refactored. implement

thautwarm committed:
Jit: make the type Verb the union of all cases. FFI.SwigGen: implemented

Commit:
GraphEngine.Jit: update asmjit binding; now it's possible to emit instructions

Commit:
GraphEngine.Jit: basic get/set jit implementation

Commit:
remove vim dump file..

Commit:
GraphEngine.Jit: remove cppsharp

Commit:
fix build

Commit:
GraphEngine.FFI/GraphEngine.Jit: implement typesystem interop structures with F#

thautwarm committed:
fix build

thautwarm committed:
fix build

Commit:
GraphEngine.Jit: establish native jit framework; ready to do some assembly generation

Commit:
Trinity.Core: refactoring. add schema change notification interface

Commit:
Merge branch 'jit' of github.com:Microsoft/GraphEngine into jit

thautwarm committed:
decoupling meta generation, compatible to swig-gen, jit-gen and any other similar generators.

thautwarm committed:
swig template

Commit:
GraphEngine.Jit: jitter wip

Commit:
Merge branch 'jit' of github.com:Microsoft/GraphEngine into jit

Commit:
update project files

Commit:
GraphEngine.Jit.Native: implement BGet BSet SGet SSet LGet LSet

thautwarm committed:
decoupling completed

Commit:
GraphEngine.Jit.Native: code refactoring on type safety; implement tsl_assign

Commit:
GraphEngine.Jit.Native: properly call exported Trinity.C interfaces

Commit:
update build script

Commit:
update build script

Commit:
fix FFI/SF builds

thautwarm committed:
a better way to handle code-gen with f#

thautwarm committed:
a better way to handle code-gen with f#

thautwarm committed:
csharp-gen implemented

thautwarm committed:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit

thautwarm committed:
Trinity.FFI.OperationFactory: MetaGen refactor again

Commit:
add jit test

Commit:
Merge branch 'jit' of github.com:Microsoft/GraphEngine into jit

thautwarm committed:
Trinity.FFI.OperationFactory.UnitTests: add tests for three implementations of meta-gen(jit, swig, csharp)

Commit:
Jit: fixed BGet/BSet: mov reg, [reg] is not valid instruction.

Commit:
Merge branch 'jit' of github.com:Microsoft/GraphEngine into jit

thautwarm committed:
Trinity.FFI.OperationFactory.Utils: format by maps -> format by associated list(list<k*v>)

Commit:
"..."

Commit:
Trinity.FFI.OperationFactory.SwigGen: refactor for python immutable data couldn't be used as references

Commit:
GraphEngine.Jit: SGet/SSet online

Commit:
test list jump, string sget sset

Commit:
Merge branch 'jit' of github.com:Microsoft/GraphEngine into jit

thautwarm committed:
Trinity.FFI.OperationFactory.UnitTests: TestJitRun, haven't passed yet

Commit:
GraphEngine.Jit: Refactor. Add int SLGetSet tests.

Commit:
Merge branch 'jit' of github.com:Microsoft/GraphEngine into jit

Commit:
GraphEngine.Jit: add StringSLGetSet test

thautwarm committed:
Trinity.FFI.Python: start new implementation

thautwarm committed:
Merge branch 'jit' of ssh://github.com/Microsoft/GraphEngine into jit

Commit:
Merge branch 'jit' of github.com:Microsoft/GraphEngine into jit

Commit:
GraphEngine.Jit: minor refactoring

Commit:
GraphEngine.Jit: some design docs

thautwarm committed:
Trinity.FFI.OperationFactory.UnitTests

thautwarm committed:
Trinity.FFI.OperationFactory: SwigCodeGen + use cell

thautwarm committed:
Merge branch 'jit' of ssh://github.com/Microsoft/GraphEngine into jit

thautwarm committed:
Trinity.FFI.OperationFactory: CodeGen(to file) finished

Commit:
GraphEngine.FFI: fix TestJitRun unit test

thautwarm committed:
Trinity.FFI: Generate swigcode to foreign end

Commit:
GraphEngine.FFI: fix build. Add GraphEngine.FFI.Metagen package

Commit:
GraphEngine.Jit/FFI: add LCount stub; minor refactor

Commit:
Trinity.FFI.Metagen: pack in necessary header file and library

Commit:
...

thautwarm committed:
Trinity.FFI.Native: unresolve external symbol: (CGetLockedCellInfo4CellAccessor, CReleaseCellLock)

thautwarm committed:
Trinity.FFI.Native: unresolve external symbol: (CGetLockedCellInfo4CellAccessor, CReleaseCellLock)

Commit:
GraphEngine.FFI.Native: link with Trinity.lib

thautwarm committed:
Trinity.FFI.Python: TSL

thautwarm committed:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit

thautwarm committed:
Trinity.FFI.Metagen/Python: fix metagen project loading. fix python build script

thautwarm committed:
Trinity.FFI.Metagen: fix codegen error. Trinity.FFI.Python: add test script

thautwarm committed:
Trinity.FFI.Python: finish test script spec

Commit:
GraphEngine.FFI: First python jit scenario is working now

thautwarm committed:
Trinity.FFI.Metagen: refactor unfinished

thautwarm committed:
Trinity.FFI: Python/Metagen: add python calling and binding spec; metagen exports deepcopied ones when object is primitive.

thautwarm committed:
Trinity.FFI.Metagen: UnitTests update.

Commit:
GraphEngine.Jit: implement LCount

thautwarm committed:
Trinity.FFI.Metagen: CodeGen fix.

thautwarm committed:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit

thautwarm committed:
Trinity.FFI.Metagen: List BGet/SGet

Commit:
Merge pull request #187 from thautwarm/jit

Trinity.FFI.Metagen:  refactor how data is accessed

thautwarm committed:
Trinity.FFI.Metagen.UnitTests: list get and count

thautwarm committed:
Trinity.FFI.Python: type system(not finished yet)

thautwarm committed:
Trinity.FFI.Python: type system(not finished yet)

thautwarm committed:
Trinity.FFI.Python: type system(not finished yet)

thautwarm committed:
Trinity.FFI.Python: type system(not finished yet)

thautwarm committed:
Trinity.FFI.Python: tsl type spec, type mapping, type system, codegen, type inference

Commit:
GraphEngine.FFI: fix Metagen TestList

thautwarm committed:
Trinity.FFI.Python: method bingding interfaces

thautwarm committed:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit

thautwarm committed:
Trinity.FFI.Python: method bingding interfaces

thautwarm committed:
Trinity.FFI.Python: data verbs implementation, mangling name consistent with Trinity.FFI.Metagen, name maker

Commit:
GraphEngine.Jit: refactoring

Commit:
Merge branch 'jit' of github.com:Microsoft/GraphEngine into jit

thautwarm committed:
Trinity.FFI.Python: binding methods from swig(or other jit middle layer) to Python runtime. const expression binding.

thautwarm committed:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit

Commit:
GraphEngine.Jit: fixing rough edges around verb mixins.
1. Mixins should not carry CellAccessor* in their GetArgs call, or we will get duplicated accessors.
2. SGet, LGet, GSGet should not intercept the GetArgs chain because there could be additional verbs after them.

Commit:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit

Commit:
GraphEngine.Jit: add verb BSame

Commit:
Merge branch 'master' into jit

Commit:
GraphEngine.Jit: add float ops, b-series ops complete

Commit:
GraphEngine.Jit: remove CppSharpCodegen

Commit:
GraphEngine.Jit: integer cmp tests passed

Commit:
...

Commit:
GraphEngine.Jit: add verb BNew. Test passed

Commit:
GraphEngine.Jit: BNew resize supported; add BHash

Commit:
GraphEngine.Jit: LContains ready

Commit:
GraphEngine.Jit: finish LInsert

Commit:
GraphEngine.Jit: LAppend finish

Commit:
GraphEngine.Jit: LRemoveAt ready

thautwarm committed:
Trinity.FFI.Python: reisen (local storage) implemented. The framework of meta generators has come true.

thautwarm committed:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit

thautwarm committed:
Trinity.FFI.Python: orm(no caching) spec implemented

thautwarm committed:
Trinity.FFI.Metagen: add list.(insert, removeAt, append)

thautwarm committed:
Trinity.FFI.Python: `use_list(type alias)` for type hinting

thautwarm committed:
Trinity.FFI.Python: `use_list(type alias)` for type hinting

Commit:
GraphEngine.Jit/GraphEngine.FFI: update LAppend, no return value; add LoadCell, SaveCell. Record cell type in BNew

Commit:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit

Commit:
GraphEngine.FFI.Metagen: cleanup

thautwarm committed:
Trinity.FFI.Native: add specifications for casting object type

thautwarm committed:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit

thautwarm committed:
Trinity.FFI.Python package: project restructured. package could be released now

Commit:
GraphEngine.Jit: all primitive ops online. all tests passed.

Commit:
Merge branch 'jit' of https://github.com/Microsoft/GraphEngine into jit
